app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Ingredient.java

```diff
@@ -0,0 +1,225 @@
+package openfoodfacts.github.scrachx.openfood.models;
+
+import org.greenrobot.greendao.DaoException;
+import org.greenrobot.greendao.annotation.Entity;
+import org.greenrobot.greendao.annotation.Generated;
+import org.greenrobot.greendao.annotation.Id;
+import org.greenrobot.greendao.annotation.Index;
+import org.greenrobot.greendao.annotation.JoinProperty;
+import org.greenrobot.greendao.annotation.Keep;
+import org.greenrobot.greendao.annotation.ToMany;
+import org.greenrobot.greendao.annotation.Unique;
+
+import java.util.List;
+
+/**
+ * Created by dobriseb on 2018.10.15.
+ */
+@Entity(
+        indexes = {
+                @Index(value = "tag DESC", unique = true)
+        }
+)
+public class Ingredient {
+    @Id(autoincrement = true)
+    private Long id;
+    @Unique
+    private String tag;
+
+    @Unique
+    private String wikiDataId;
+
+    @ToMany(joinProperties = {
+        @JoinProperty(name = "tag", referencedName = "ingredientTag")
+    })
+    private List<IngredientName> names;
+
+    @ToMany(joinProperties = {
+        @JoinProperty(name = "tag", referencedName = "parentTag")
+    })
+    private List<IngredientsRelation> parents;
+
+    @ToMany(joinProperties = {
+        @JoinProperty(name = "tag", referencedName = "childTag")
+    })
+    private List<IngredientsRelation> children;
+/** Used to resolve relations */
+@Generated(hash = 2040040024)
+private transient DaoSession daoSession;
+/** Used for active entity operations. */
+@Generated(hash = 942581853)
+private transient IngredientDao myDao;
+
+@Generated(hash = 332885183)
+public Ingredient(Long id, String tag, String wikiDataId) {
+    this.id = id;
+    this.tag = tag;
+    this.wikiDataId = wikiDataId;
+}
+@Keep
+public Ingredient(String tag, List<IngredientName> names, List<IngredientsRelation> parents, List<IngredientsRelation> children, String wikiDataId) {
```

Comment:
Please, could you reformat this file ( no spaces currently)

Reply:
Done... if I understood.